torch/optim/asgd.py

```diff
@@ -261,22 +260,22 @@ def _view_complex_as_real(tensor_list):
         grads = torch._foreach_add(grads, params, alpha=weight_decay)
 
     # decay term
-    eta = etas[0].item()
+    eta = etas[0]
```

Comment:
Isn't the original code plain wrong?? All the params are updates with a single eta from the 0th value.

Reply:
So I think the reason this works is if all of the etas are on the same device, etas[:] will all hold the same value. eta is only a funciton lambd, lr, step, and alpha which are constants per group

Reply:
I'm also fine with fixing it though

Reply:
We should fix it for sure. But this PR is not the right place for this as it need proper testing/etc